### PR TITLE
Creating tests long etl

### PR DESCRIPTION
**Pull request by YakMan101**
Added more test cases to test transform.py functionality. Also added some bug fixes for transform.py including:

- rounding numerical values
- Edge case where if last_watered column contains just None it gets recorded as 1 times watered instead of 0.
- No plant_name column, just plant_id

### CODE REVIEW

File: long_term/test_transform_long.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+from unittest.mock import patch
+
+import pytest
+import pandas as pd
+
+from transform_long import create_summary, round_numerical_columns
+
+
+@pytest.mark.parametrize("input_data, columns, round_dp, expected_output", [
+    (pd.DataFrame({"A": [1.1234, 2.5678, None], "B": [3.1415, None, 2.7182]}), ["A", "B"], 2,
```
Comment (Leon-JS02) [inline long_term/test_transform_long.py:13]: I like these parametrised tests.

File: long_term/test_transform_long.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+def test_round_numerical_columns(input_data, columns, round_dp, expected_output):
+    result = round_numerical_columns(input_data, columns, round_dp)
+    result = result.fillna('N/A')
+    expected_output = expected_output.fillna('N/A')
+
+    print(result, expected_output)
+    pd.testing.assert_frame_equal(result, expected_output)
+
+
+class TestCreateSummary:
```
Comment (Leon-JS02) [inline long_term/test_transform_long.py:38]: Nice use of OOP.

**Review by Leon-JS02 — APPROVED** [review]
Nice test files with appropriate edge cases considered.

**Review by mehrdadhalali — APPROVED** [review]
